docs/fundamentals/databases.md

```diff
@@ -0,0 +1,58 @@
+| `compWriteMeter`     | Meter for measuring the data written during compaction                  |
+| `writeDelayNMeter`   | Meter for measuring the write delay number due to database compaction   |
+| `writeDelayMeter`    | Meter for measuring the write delay duration due to database compaction |
+| `diskSizeGauge`      | Gauge for tracking the size of all the levels in the database           |
+| `diskReadMeter`      | Meter for measuring the effective amount of data read                   |
+| `diskWriteMeter`     | Meter for measuring the effective amount of data written                |
+| `memCompGauge`       | Gauge for tracking the number of memory compaction                      |
+| `level0CompGauge`    | Gauge for tracking the number of table compaction in level0             |
+| `nonlevel0CompGauge` | Gauge for tracking the number of table compaction in non0 level         |
+| `seekCompGauge`      | Gauge for tracking the number of table compaction caused by read opt    |
+
+
+## Freezer/ancients
+
+Older segments of the chain are moved out of the fast LevelDB database and into a freezer database the is less performant. Nodes rarely need to access these files so IO speed is less important and the bulk of the chain data can be stored on a cheaper HDD. Once blocks pass some threshold age (128 blocks behind the head by default) the block and receipt data is flattened and saved as a raw binary blob of data along with an index entry file used for identification. This flat file format allows Geth to capitalize on the operating system's built-in memory manager and file cacher which in turn allows Geth to maximize the available memory without hogging it from other processes. This is good for overall system stability and also reduces the strain on Go's garbage collector and reduces the total amount of memory used by Geth. 
+
+Geth also tracks some basic metrics relating to the ancients database that can be monitored:
+
+| metric       | function                                                   |
+| ------------ | ---------------------------------------------------------- |
+| `readMeter`  | Meter for measuring the effective amount of data read      |
+| `writeMeter` | Meter for measuring the effective amount of data written   |
+| `sizeGauge`  | Gauge for tracking the combined size of all freezer tables |
+
+
+The ancients data is saved entirely separately from the fast-access recent data, meaning it can be stored in a  different location. The default location for the ancient chain segments is inside the `chaindata` directory, which is inside `datadir`, but it can be defined by passing `--datadir.ancient <path>` to Geth on startup. This path can be to a slow HDD disk, permitting people to run Ethereum nodes without requiring a huge SSD. The ancient can also be moved later by manually copying the directory to a new location and then starting Geth passing the new path to `--datadir.ancient`.
```

Comment:
I would appreciate if we can advertise our freezer design a bit.

The complexity of read operation in freezer is O(1), which only involves:
- a read for index item(6 bytes)
- a read for data(depends on the size of data)
And that's all. Freezer should still be performant enough on slow HDD
because of this design.